src/server_config.c

```diff
@@ -3135,9 +3192,15 @@ config_unix(const struct lyd_node *node, enum nc_operation parent_op, struct nc_
         endpt->opts.unix->gid = (gid_t)-1;
     }
 
-    /* config path */
-    NC_CHECK_RET(nc_lyd_find_child(node, "path", 1, &n));
-    NC_CHECK_RET(config_unix_path(n, op, endpt));
+    /* config mandatory unix socket path choice => only one of them can be present */
+    NC_CHECK_RET(nc_lyd_find_child(node, "socket-path", 0, &socket_path));
+    NC_CHECK_RET(nc_lyd_find_child(node, "hidden-path", 0, &hidden_path));
+    if (socket_path) {
+        NC_CHECK_RET(config_unix_socket_path(socket_path, op, endpt));
+    } else {
+        assert(hidden_path);
```

Comment:
If data validation is not performed before this code, there can still be invalid data and those should not cause a failed assert/crash. At least change the `else` into `else if (hidden_path)`.